### PR TITLE
looper_scheduler worksize and bug fixes

### PR DESCRIPTION
**Pull request by bghgary**

### CODE REVIEW

File: Source/Android/arcana/threading/task_schedulers.h [hunk 1]
```diff
@@ -27,6 +27,7 @@ namespace arcana
         } threadpool_scheduler{};
     }
 
+    template<size_t WorkSize>
```
Comment (ryantrem) [inline Source/Android/arcana/threading/task_schedulers.h]: Elsewhere we default this template param to ``stdext::InplaceFunctionDefaultCapacity`. Should we do that here too?

File: Source/Android/arcana/threading/task_schedulers.h [hunk 2]
```diff
@@ -66,23 +61,22 @@ namespace arcana
...
         {
+            reset();
+
             m_looper = other.m_looper;
             m_fd[0] = other.m_fd[0];
             m_fd[1] = other.m_fd[1];
 
             other.m_looper = nullptr;
-            other.m_fd[0] = 0;
-            other.m_fd[1] = 0;
```
Comment (ryantrem) [inline Source/Android/arcana/threading/task_schedulers.h]: Why don't we need these any more? What happens if `other` is invoked after `this` has taken over?

Comment (bghgary) [reply Source/Android/arcana/threading/task_schedulers.h]: Right, good point. I should keep this.

File: Source/Android/arcana/threading/task_schedulers.h (hunk 1 above)
Comment (bghgary) [reply Source/Android/arcana/threading/task_schedulers.h]: I was thinking the same thing, but not sure if we should. The only place that uses that is the inplace_function class itself. Dispatcher, for example, doesn't set a default.

Comment (ryantrem) [reply Source/Android/arcana/threading/task_schedulers.h]: Oh interesting, I didn't remember that. Ok then I guess what you have is consistent.

File: Source/Android/arcana/threading/task_schedulers.h (hunk 2 above)
Comment (bghgary) [reply Source/Android/arcana/threading/task_schedulers.h]: Apparently -1 is an invalid file descriptor as that's what gets returned by `open` for example when it fails.

**Pull request merged**